packages/lit-html/src/lit-html.ts

```diff
@@ -104,7 +104,7 @@ const comment2Regex = />/g;
  *    * (') then any non-(')
  */
 const tagRegex = new RegExp(
-  `>|${SPACE_CHAR}(${NAME_CHAR}+)(${SPACE_CHAR}*=${SPACE_CHAR}*(?:${ATTR_VALUE_CHAR}|("|')|$))`,
+  `>|${SPACE_CHAR}(${NAME_CHAR}+)(${SPACE_CHAR}*=${SPACE_CHAR}*(?:${ATTR_VALUE_CHAR}|("|')|))`,
```

Comment:
The comments for these regex's should very clearly describe the capturing groups since they are critical to the code below,

Reply:
Is it true that `ATTR_VALUE_CHAR` can include `/`?
Is it true that ` NAME_CHAR` can include `<`?
Are there tests for these?

Reply:
Looks like it: `<input foo<bar=a/b>` is valid. Can add tests in a follow on PR.

Reply:
These parts should cover that:
```
 *    * Any character except space, ('), ("), "<", ">", "=", (`), or
 *    * (") then any non-("), or	 *    * (") then any non-("), or
 *    * (') then any non-(')	 *    * (') then any non-(')
```

They're pretty wide-open in terms of what's allowed.

I'll add that test.

Reply:
This name doesn't seem to fit since this can match without ending the tag, right?

`tagOrAttrEndRegex`?

Reply:
I guess it depends on the definition of the state. What I mean here is that this regex detects the end of the "tag state", which can be the tag end, or an attribute.

Reply:
Still think this name is weird, but it's hard to name.